test/FunctionalTests/Microsoft.OData.Client.Tests/TestHttpWebRequestMessage.cs

```diff
@@ -0,0 +1,60 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="TestHttpWebRequestMessage.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.OData.Client.Tests
+{
+    public class TestHttpWebRequestMessage : HttpWebRequestMessage
+{
+	private readonly Func<Stream> getResponseStream;
+	private readonly int statusCode;
+	private readonly IDictionary<string, string> headers;
+
+	public TestHttpWebRequestMessage(DataServiceClientRequestMessageArgs args)
+		: this(args, new Dictionary<string, string>(), 200, () => new MemoryStream(Encoding.UTF8.GetBytes("")))
+	{
+	}
+
+	public TestHttpWebRequestMessage(DataServiceClientRequestMessageArgs args, IDictionary<string, string> headers, Func<Stream> getResponseStream)
+		: this(args, headers, 200, getResponseStream)
+    {
+    }
+
+	public TestHttpWebRequestMessage(DataServiceClientRequestMessageArgs args, IDictionary<string, string> headers, int statusCode, Func<Stream> getResponseStream)
+		: base(args)
+	{
+		this.headers = headers;
+		this.statusCode = statusCode;
+		this.getResponseStream = getResponseStream;
+	}
+
+#if (NETCOREAPP1_0 || NETCOREAPP2_0)

```

Comment:
Is this check necessary? What about core 3+?

Reply:
@odero Maybe not... I'll remove it and see how the tests behave